mlir/dialects/linalg.py

```diff
@@ -0,0 +1,291 @@
+""" Implementation of the Linalg dialect. """
+
+import inspect
+import sys
+import mlir.astnodes as mast
+from mlir.dialect import Dialect, DialectOp, is_op
+from dataclasses import dataclass
+from typing import Optional, List
+
+
+@dataclass
+class LinalgBatchMatmul(DialectOp):
+    a_id: mast.SsaId
+    b_id: mast.SsaId
+    a_type: mast.Type
+    b_type: mast.Type
+    c_id: mast.SsaId
+    c_type: mast.Type
+    out_type: Optional[mast.Type] = None
+
+    _syntax_ = [("linalg.batch_matmul"
+                 " ins( {a_id.ssa_id} , {b_id.ssa_id} : {a_type.type} , {b_type.type} )"
+                 " outs( {c_id.ssa_id} : {c_type.type} )"),
+                ("linalg.batch_matmul"
+                 " ins( {a_id.ssa_id} , {b_id.ssa_id} : {a_type.type} , {b_type.type} )"
+                 " init( {c_id.ssa_id} : {c_type.type} ) -> {out_type.type}")]
+
+
+@dataclass
+class LinalgConvW(DialectOp):
+    in_id: mast.SsaId
+    filter_id: mast.SsaId
+    in_type: mast.Type
+    filter_type: mast.Type
+    out_id: mast.SsaId
+    out_type: mast.Type
+
+    _syntax_ = [("linalg.conv_1d"
+                 " ins( {in_id.ssa_id} , {filter_id.ssa_id} : {in_type.type} , {filter_type.type} )"
+                 " outs( {out_id.ssa_id} : {out_type.type} )")]
+
+
+@dataclass
+class LinalgConvHW(DialectOp):
+    in_id: mast.SsaId
+    filter_id: mast.SsaId
+    in_type: mast.Type
+    filter_type: mast.Type
+    out_id: mast.SsaId
+    out_type: mast.Type
+
+    _syntax_ = [("linalg.conv_2d"
+                 " ins( {in_id.ssa_id} , {filter_id.ssa_id} : {in_type.type} , {filter_type.type} )"
+                 " outs( {out_id.ssa_id} : {out_type.type} )")]
+
+
+@dataclass
+class LinalgConvDHW(DialectOp):
```

Comment:
```suggestion
class LinalgConv3d(DialectOp):
```

Reply:
I could go either way on this. The reason I chose the DHW suffix instead of 3d was to match with the AST class on the C++ side: https://mlir.llvm.org/docs/Dialects/Linalg/#linalgconv_3d-mlirlinalgconvdhwop.

Reply:
Let me know if you have a strong preference towards one, happy to change it. Thought keeping consistency with the naming on the mlir-end could be helpful.

Reply:
OK, I wasn't aware of the naming convention in the C++ AST class. It makes much more sense now.